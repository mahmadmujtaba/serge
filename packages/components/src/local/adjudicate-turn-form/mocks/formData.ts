import { AdjudicateTurnFormPopulate } from "@serge/custom-types"

const formData: AdjudicateTurnFormPopulate = {
  contactId: 'C234',
  status: [
    {
      name: 'Fishing',
      mobile: false
    },
    {
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }
  ],
  speed: [10, 20, 30],
  visibleTo: [
    {
      name: 'Blue Force',
      colour: '#69c'
    },
    {
      name: 'Red Force',
      colour: '#f00'
    },
    {
      name: 'White Force',
      colour: '#fff'
    }
  ],
  condition: ['Working', 'Disabled', 'Immobile', 'Destroyed']
}

export default formData
