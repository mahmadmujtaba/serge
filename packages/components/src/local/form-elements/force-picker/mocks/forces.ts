const forces = [
  {
    name: 'Blue Force',
    colour: '#69c',
    selected: true
  },
  {
    name: 'Red Force',
    colour: '#f00',
    selected: false
  },
  {
    name: 'White Force',
    colour: '#fff',
    selected: false
  }
]

export default forces
