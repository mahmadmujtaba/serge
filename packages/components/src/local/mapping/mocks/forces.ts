export const forces = [
    {
      color: '#FCFBEE',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'White',
      overview: 'Umpire force.',
      roles: [
        {
          control: true,
          isInsightViewer: true,
          isObserver: true,
          name: 'Game Control',
          password: 'p2311'
        }
      ],
      umpire: true,
      uniqid: 'umpire'
    },
    {
      assets: [
        {
          name: 'HMS DEVONSHIRE',
          contactId: 'C043',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'frigate',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00001'
        },
        {
          name: 'STING',
          contactId: 'C072',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'helicopter',
          position: 'P01',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00002'
        },
        {
          name: 'P8',
          contactId: 'C076',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'fixed-wing-aircraft',
          position: 'P03',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00012'
        },
        {
          name: 'EDDYSTONE PT',
          contactId: 'C012',
          condition: 'Full capability',
          perceptions: {
            Red: {
              force: 'Blue',
              type: ''
            }
          },
          platformType: 'merchant-vessel',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00003'
        }
      ],
      color: '#00F',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Blue',
      overview: 'Blue force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p5543'
        }
      ],
      umpire: false,
      uniqid: 'Blue'
    },
    {
      assets: [
        {
          name: 'TARRADA',
          contactId: 'C065',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'fishing-vessel',
          position: 'N07',
          status: {
            state: 'Transiting',
            speedKts: 10
          },
          uniqid: 'a0pra000100'
        },
        {
          name: 'WBIED-1',
          contactId: 'C108',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T01',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra020102'
        },
        {
          name: 'WBIED-2',
          contactId: 'C118',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T03',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra130102'
        },
        {
          name: 'Osa',
          contactId: 'C003',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: ''
            }
          },
          platformType: 'fast-attack-craft',
          position: 'M05',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra160102'
        },
        {
          name: 'Yevgenya',
          contactId: 'C313',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: 'fast-attack-craft'
            }
          },
          platformType: 'mcmv',
          position: 'M04',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra170102'
        },
        {
          name: 'Coastal Radar C',
          contactId: 'C476',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'coastal-radar-site',
          position: 'Q12',
          status: {
            state: 'Inactive'
          },
          uniqid: 'a0prc000103'
        }
      ],
      color: '#F00',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Red',
      overview: 'Red force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p3244'
        }
      ],
      umpire: false,
      uniqid: 'Red'
    }
  ]


export default forces
