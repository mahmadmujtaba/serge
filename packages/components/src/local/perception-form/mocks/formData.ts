import { PerceptionFormData } from '@serge/custom-types'

const formData: PerceptionFormData = {
  populate: {
    perceivedForces: [
      {
        name: 'Blue Force',
        colour: '#69c'
      },
      {
        name: 'Red Force',
        colour: '#f00'
      },
      {
        name: 'White Force',
        colour: '#fff'
      }
    ],
    perceivedTypes: [
      { uniqid: 'aa', name:'Fishing vessel'},
      { uniqid: 'a1', name:'torpedo'},
      { uniqid: 'a2', name:'Boghammer'},
      { uniqid: 'a3', name:'Frigate'}
    ]
  },
  values: {
    perceivedNameVal: 'Test',
    perceivedForceVal: 'blue-force',
    perceivedTypeId: 'a3',
    assetId: 'c00012',
    iconURL: 'frigate.svg'
  }
}

export default formData
