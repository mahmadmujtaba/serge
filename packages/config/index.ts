/* Pass the exports from 'src' to other TS enabled packages */

export * from './src'