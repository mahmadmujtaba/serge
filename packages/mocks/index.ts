/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as MessagesMock } from './messages.mock'
export { default as WargameMock } from './wargame.mock'