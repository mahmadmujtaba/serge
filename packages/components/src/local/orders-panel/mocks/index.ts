import { ADJUDICATION_PHASE } from '@serge/config'

const data = {
  selectedForce: 'blue',
  phase: ADJUDICATION_PHASE
}

export default data
