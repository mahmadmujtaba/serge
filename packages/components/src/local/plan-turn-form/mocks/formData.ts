import { COMMODITY_TYPE_NUMBER, COMMODITY_VALUE_NUMBER } from '@serge/config'
import { PlanTurnFormData } from '@serge/custom-types'

const formData: PlanTurnFormData = {
  populate: {
    status: [{
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }],
    speed: [10, 20, 30],
    attributes: [{ commId: 'comm_a', commType: COMMODITY_TYPE_NUMBER, name: 'Fuel', editableByPlayer: false },
      { commId: 'comm_b', commType: COMMODITY_TYPE_NUMBER, name: 'People', editableByPlayer: false },
      { commId: 'comm_c', commType: COMMODITY_TYPE_NUMBER, name: 'Water', units: 'litres', editableByPlayer: false }]
  },
  values: {
    statusVal: {
      name: 'Transiting',
      mobile: true
    },
    speedVal: 10,
    turnsVal: 5,
    condition: 'Working',
    attributes: [{ commId: 'comm_a', commType: COMMODITY_VALUE_NUMBER, value: 12 },
      { commId: 'comm_b', commType: COMMODITY_VALUE_NUMBER, value: 213 },
      { commId: 'comm_c', commType: COMMODITY_VALUE_NUMBER, value: 12450 }]
  }
}

export default formData
