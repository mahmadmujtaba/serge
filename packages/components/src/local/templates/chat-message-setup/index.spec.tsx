/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import ChatMessageSetup from './index'

import { ChatMessagesMock } from '@serge/mocks'
import moment from 'moment-timezone'
import { TurnFormats } from '@serge/config'

const force = {
  name: 'blue',
  color: '#6699cc',
  icon: '',
  iconURL: ''
}

const createNodeMock = (el: any): HTMLTextAreaElement | null => {
  if (el.type === 'textarea') {
    return document.createElement('textarea')
  }
  return null
}

describe('ChatMessageSetup page:', () => {
  it('renders correctly', () => {
    moment.tz.setDefault('Etc/UTC')
    const tree = renderer
      .create(
        <ChatMessageSetup turnNumber={1} userId={'CO'} forceName={'Blue'} isUmpire={true} chatChannel={ChatMessagesMock}
          selectedForce={force} selectedRoleName={'CO'} turnPresentation={TurnFormats.Natural} selectedRole={'u3456'}/>,
        { createNodeMock }
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
