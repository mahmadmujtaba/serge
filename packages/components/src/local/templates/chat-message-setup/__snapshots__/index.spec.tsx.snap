// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessageSetup page: renders correctly 1`] = `
Array [
  <div>
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <div
        className="forces-in-channel"
      >
        <img
          alt=""
          className="force-indicator role-icon"
          src="./images/default_img/forceDefault.png"
          style={
            Object {
              "backgroundColor": "#0F0",
              "borderColor": "#0F0",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-3 messages-list"
      style={
        Object {
          "height": "calc(100vh - 280px)",
        }
      }
    >
      <div
        className="MuiBox-root MuiBox-root-4"
      >
        <p
          className="turn-marker"
        >
          Turn 
          0
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-5"
      >
        <div
          className="MuiBox-root MuiBox-root-6"
          style={
            Object {
              "float": "left",
            }
          }
        >
          <div
            className="chat-message-wrapper 
      chat-message-wrapper__other"
            style={
              Object {
                "borderLeftColor": "#FCFBEE",
              }
            }
          >
            <div
              className="message-text"
            >
              test

            </div>
            <div
              className="MuiBox-root MuiBox-root-7"
            >
              <span>
                <span
                  className="icon-private"
                >
                  <svg
                    className="svg-inline--fa fa-user-secret"
                  />
                </span>
                <span
                  className="label-private"
                >
                  Private message:
                </span>
              </span>
              <div
                className="MuiBox-root MuiBox-root-8"
              >
                <div
                  className="MuiChip-root makeStyles-root-9 makeStyles-root-12 MuiChip-sizeSmall makeStyles-sizeSmall-10"
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                >
                  <span
                    className="MuiChip-label makeStyles-label-11 makeStyles-label-13 MuiChip-labelSmall"
                  >
                    Game Control
                  </span>
                </div>
                <span
                  className="info-body info-body__other"
                >
                  15:08
                </span>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-14 private"
            >
              <div
                className="paragraph"
              >
                Private message here
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-15"
      >
        <p
          className="turn-marker"
        >
          Turn 
          1
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-16"
      >
        <div
          className="MuiBox-root MuiBox-root-17"
          style={
            Object {
              "float": "right",
            }
          }
        >
          <div
            className="chat-message-wrapper 
      chat-message-wrapper__owner"
            style={
              Object {
                "borderRightColor": "#1989dc",
              }
            }
          >
            <div
              className="message-text"
            >
              Lorem ipsum do lor sit amet adipiscius elit

            </div>
            <div
              className="MuiBox-root MuiBox-root-18"
            >
              <div
                className="MuiBox-root MuiBox-root-19"
              >
                <div
                  className="MuiChip-root makeStyles-root-9 makeStyles-root-20 MuiChip-sizeSmall makeStyles-sizeSmall-10"
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                >
                  <span
                    className="MuiChip-label makeStyles-label-11 makeStyles-label-21 MuiChip-labelSmall"
                  >
                    CO
                  </span>
                </div>
                <span
                  className="info-body info-body__owner"
                >
                  15:08
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-22 chat-container"
  >
    <section
      className="input-container disable-offset fullwidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        onKeyDown={[Function]}
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-root-23 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline makeStyles-multiline-25 MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input makeStyles-input-24 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
            disabled={false}
            name=""
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="type the text"
            required={false}
            rows={2}
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
            value=""
          />
          <textarea
            aria-hidden={true}
            className="MuiInputBase-input MuiFilledInput-input makeStyles-input-24 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
            readOnly={true}
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "transform": "translateZ(0)",
                "visibility": "hidden",
              }
            }
            tabIndex={-1}
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root icon-send MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4.01 6.03l7.51 3.22-7.52-1 .01-2.22m7.5 8.72L4 17.97v-2.22l7.51-1M2.01 3L2 10l15 2-15 2 .01 7L23 12 2.01 3z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </section>
    <div
      className="MuiBox-root MuiBox-root-30"
    >
      <div
        className="MuiBox-root MuiBox-root-31"
        onClick={[Function]}
      >
        <span
          className="private-badge-wrapper"
        >
          <span>
            <svg
              className="svg-inline--fa fa-paperclip"
            />
          </span>
          <span
            className="icon-private"
          >
            <svg
              className="svg-inline--fa fa-user-secret"
            />
          </span>
          <span
            className="label-private"
          >
            Private message
          </span>
        </span>
      </div>
      <div
        className="MuiBox-root MuiBox-root-32 wrap-detail wrap-detail-opened"
      >
        <section
          className="input-container disable-offset fullwidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            onKeyDown={[Function]}
          >
            <div
              className="MuiInputBase-root MuiFilledInput-root makeStyles-root-23 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline makeStyles-multiline-25"
              onClick={[Function]}
            >
              <textarea
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input makeStyles-input-24 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
                disabled={false}
                name=""
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                rows={2}
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
                value=""
              />
              <textarea
                aria-hidden={true}
                className="MuiInputBase-input MuiFilledInput-input makeStyles-input-24 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
                readOnly={true}
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "visibility": "hidden",
                  }
                }
                tabIndex={-1}
              />
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>,
]
`;
